src/test/common/test_random.cc

```diff
@@ -0,0 +1,66 @@
+// -*- mode:C; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*- 
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2011 New Dream Network
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License version 2, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+#include <sstream>
+
+#include "include/random.h"
+
+#include "gtest/gtest.h"
+
+TEST(util, test_random)
+{
+ /* The intent of this test is not to formally test random number generation, but rather to
+ casually check that "it works" and catch regressions: */
+
+ {
+ auto a = ceph::util::generate_random_number();
+ auto b = ceph::util::generate_random_number();
+
+ // Technically, this can still collide and cause a false test failure, but let's be optimistic:
+ if(std::numeric_limits<int>::max() > 32767)
+  ASSERT_NE(a, b);
+ }
+
+ {
+ auto a = ceph::util::generate_random_number(std::numeric_limits<int>::max(), 1);
+ auto b = ceph::util::generate_random_number(std::numeric_limits<int>::max(), 1);
+
+ if(std::numeric_limits<int>::max() > 32767)
+  ASSERT_NE(a, b);
```

Comment:
An additional check might be for not-zero. Maybe it would be more interesting to narrow the range? I don't want this testing the RNG itself.